### ISSUE TITLE
not able to insert array columns

### ISSUE THREAD

**Issue by srinireddya** [spring-projects/spring-data-r2dbc#119]
I have a table defined as mentioned below

```sql
CREATE TABLE tablea (
	id serial primary key ,
	text_array text[],
	int_array int[]
) ;
```

Have a entity object defined as 

```java
@AllArgsConstructor
@NoArgsConstructor
@EntityScan
@Getter @Setter 
@Table("tablea")
public class Tablea {
    @Id
    private Long id;
    private List<String> textArray;
    private List<Integer> intArray = Arrays.asList(1,2,3);
```

When i insert the data both the arrays are inserted as null. If i manually insert them and try to retrieve it is working fine

**Comment by mp911de** [comment]
Thanks for report. May I assume this is Postgres? Please let us know which versions you're using.

**Comment by srinireddya** [comment]
This is postgres and i am using version 10.7

**Comment by JoseLion** [comment]
Hi @mp911de! I'm having a similar issue, I'm using `r2dbc-postgresql` to create a `PostgresqlConnectionFactory` and everything was working just fine until I added a `text[]` column mapped as `String[]` (r2dbc-postgresql documentation says it's the supported mapping type). Then when I tried to retrieve the data I go the following exception:
```
Caused by: java.lang.IllegalArgumentException: Cannot decode value of type java.lang.Object
	at io.r2dbc.postgresql.codec.DefaultCodecs.decode(DefaultCodecs.java:96) ~[r2dbc-postgresql-1.0.0.M7.jar:na]
	at io.r2dbc.postgresql.PostgresqlRow.get(PostgresqlRow.java:83) ~[r2dbc-postgresql-1.0.0.M7.jar:na]
	at io.r2dbc.spi.Row.get(Row.java:46) ~[r2dbc-spi-1.0.0.M7.jar:na]
	at org.springframework.data.r2dbc.convert.MappingR2dbcConverter$RowParameterValueProvider.getParameterValue(MappingR2dbcConverter.java:426) ~[spring-data-r2dbc-1.0.0.M2.jar:1.0.0.M2]
	... 81 common frames omitted
```

I think I actually found the source of the problem which I explained in [this issue](https://github.com/r2dbc/r2dbc-postgresql/issues/113) from `r2dbc-postgresql` repository (at first I though the issue was on their side 😝). Apparently the type needs to be passed to the `get(..)` method of the `Row` interface, otherwise the type is always passed as `java.lang.Object`, and `r2dbc-postgresql` has a validation that checks if the type is an array. Hence `Object.class.isArray()` will always be false.

Please let me know what you think. If this makes sense I will be happy to open a PR with the fix 🙂

**Comment by mp911de** [comment]
We're calling the `.get(identifier)` method as Spring Data isn't opinionated about the type during retrieval from the store. In R2DBC, `.get(identifier)` corresponds with `.get(identifier, Object.class)` resulting in that the driver needs to return a value that is assignable to `Object`.

TL;DR `.get(identifier)` instructs the driver to return the natively used type and the driver has to make sure that it returns the right type. Therefore, I conclude that r2dbc/r2dbc-postgresql#113 is a bug.

**Comment by JoseLion** [comment]
@mp911de thanks a lot for the response and explanation I really appreciate it 🙂
I understand you point and I agree with you! I think that the portion of code in `r2dbc-postgresql` can work without the `type.isArray()` validation, so I'll go ahead and try to open a PR for https://github.com/r2dbc/r2dbc-postgresql/issues/113. I'll post any update just to keep everyone in the loop 🙂

**Comment by mp911de** [comment]
The upstream issue is solved, so closing this ticket.
